server/strategy.go

```diff
@@ -625,48 +641,63 @@ func cleanOps(c *cke.Cluster, nf *NodeFilter) (ops []cke.Operator) {
 	return ops
 }
 
-func rebootOps(c *cke.Cluster, entry *cke.RebootQueueEntry, nf *NodeFilter) []cke.Operator {
-	if entry == nil {
+func rebootOps(c *cke.Cluster, rqEntries []*cke.RebootQueueEntry, newlyDrained []*cke.RebootQueueEntry, drainCompleted []*cke.RebootQueueEntry, drainTimedout []*cke.RebootQueueEntry, rebootDequeued []*cke.RebootQueueEntry, nf *NodeFilter) (ops []cke.Operator) {
+	if len(rqEntries) == 0 {
 		return nil
 	}
-	if entry.Status == cke.RebootStatusCancelled {
-		return []cke.Operator{op.RebootDequeueOp(entry.Index)}
-	}
-	if len(c.Reboot.Command) == 0 {
+	if len(c.Reboot.RebootCommand) == 0 {
```

Comment:
I suggest checking the settings with `validateReboot`.
Because we will notice the problem earlier.

https://github.com/cybozu-go/cke/blob/0101d79b51f6662d91f98159e4f161d55339f015/cluster.go#L460

Reply:
- If the cluster administrator does not need rebooting support, those fields are empty.
- During CKE upgrade, those fields may be empty.

So, checking those fields in `validateReboot` may cause a problem.

Reply:
I see.